internal/dashboard/handler/helper.go

```diff
@@ -47,3 +54,53 @@ func loggingCreateOBTenantParam(param *param.CreateOBTenantParam) {
 		WithField("Source", param.Source).
 		Infof("Create OBTenant param")
 }
+
+type OdcBastionData struct {
+	AccountVerifyToken string `json:"accountVerifyToken"`
+	Type               string `json:"type"`
+	UnionDbUser        string `json:"unionDbUser"`
+	Host               string `json:"host"`
+	Password           string `json:"password"`
+	Port               int    `json:"port"`
+}
+
+type OdcBastion struct {
+	Action  string         `json:"action"`
+	Data    OdcBastionData `json:"data"`
+	Encrypt bool           `json:"encrypt"`
+}
+
+func generateOdcParam(c *gin.Context, host, user, passwd string) (string, error) {
+	session := sessions.Default(c)
+	var username string
+	if session.Get("username") == nil {
+		username = "user"
+	} else {
+		username = session.Get("username").(string)
+	}
+
+	token := auth.GenerateAuthToken(&auth.AuthUser{
+		Username: username,
+		Nickname: username,
+	})
+
+	odcBastionData := OdcBastionData{
+		AccountVerifyToken: token.String(),
+		Type:               "OB_MYSQL",
+		UnionDbUser:        user,
+		Host:               host,
+		Password:           passwd,
+		Port:               2881,
+	}
+
+	odcBastion := OdcBastion{
```

Comment:
is this only for unencrypted data

Reply:
Yes, the encryption algorithm (`SecureRandom` and `KeyGenerator`) was written in Java and there is no corresponding implementation in Go.

Reply:
OK